src/InstrumentationEngine/ProfilerManagerWrapper.cpp

```diff
@@ -0,0 +1,197 @@
+ */
+HRESULT CProfilerManagerWrapper::GetLoggingInstance(_Out_ IProfilerManagerLogging** ppLogging)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IProfilerManagerLogging> pProfilerManagerLogging = (IProfilerManagerLogging*)this;
+    *ppLogging = pProfilerManagerLogging.Detach();
+    return S_OK;
+}
+
+HRESULT CProfilerManagerWrapper::SetLoggingHost(_In_opt_ IProfilerManagerLoggingHost* pLoggingHost)
+{
+    return m_pProfilerManager->SetLoggingHost(pLoggingHost);
+}
+
+HRESULT CProfilerManagerWrapper::GetAppDomainCollection(_Out_ IAppDomainCollection** ppAppDomainCollection)
+{
+    return m_pProfilerManager->GetAppDomainCollection(ppAppDomainCollection);
+}
+
+
+HRESULT CProfilerManagerWrapper::CreateSignatureBuilder(_Out_ ISignatureBuilder** ppSignatureBuilder)
+{
+    return m_pProfilerManager->CreateSignatureBuilder(ppSignatureBuilder);
+}
+
+HRESULT CProfilerManagerWrapper::GetInstrumentationMethod(_In_ REFGUID cslid, _Out_ IUnknown** ppUnknown)
+{
+    return m_pProfilerManager->GetInstrumentationMethod(cslid, ppUnknown);
+}
+
+HRESULT CProfilerManagerWrapper::RemoveInstrumentationMethod(_In_ IInstrumentationMethod* pInstrumentationMethod)
+{
+    return m_pProfilerManager->RemoveInstrumentationMethod(pInstrumentationMethod);
+}
+
+HRESULT CProfilerManagerWrapper::AddInstrumentationMethod(
+    _In_ BSTR bstrModulePath,
+    _In_ BSTR bstrName,
+    _In_ BSTR bstrDescription,
+    _In_ BSTR bstrModule,
+    _In_ BSTR bstrClassGuid,
+    _In_ DWORD dwPriority,
+    _Out_ IInstrumentationMethod** ppInstrumentationMethod)
+{
+    return m_pProfilerManager->AddInstrumentationMethod(
+        bstrModulePath,
+        bstrName,
+        bstrDescription,
+        bstrModule,
+        bstrClassGuid,
+        dwPriority,
+        ppInstrumentationMethod);
+}
+
+// IProfilerManager2 Methods
+HRESULT CProfilerManagerWrapper::DisableProfiling()
+{
+    return m_pProfilerManager->DisableProfiling();
+}
+
+HRESULT CProfilerManagerWrapper::ApplyMetadata(_In_ IModuleInfo* pMethodInfo)
+{
+    return m_pProfilerManager->ApplyMetadata(pMethodInfo);
+}
+
+// IProfilerManager3 Methods
+HRESULT CProfilerManagerWrapper::GetApiVersion(_Out_ DWORD* pApiVer)
+{
+    return m_pProfilerManager->GetApiVersion(pApiVer);
+}
+
+// IProfilerManager4 Methods
+HRESULT CProfilerManagerWrapper::GetGlobalLoggingInstance(_Out_ IProfilerManagerLogging** ppLogging)
+{
+    return m_pProfilerManager->GetGlobalLoggingInstance(ppLogging);
+}
+
+
+// IProfilerManager5 Methods
+HRESULT CProfilerManagerWrapper::IsInstrumentationMethodRegistered(_In_ REFGUID clsid, _Out_ BOOL* pfRegistered)
+{
+    return m_pProfilerManager->IsInstrumentationMethodRegistered(clsid, pfRegistered);
+}
+
+// IProfilerManagerLogging Methods
+/*
+ * Inject Instrumentation Method ClassId
+ */
+HRESULT CProfilerManagerWrapper::LogMessage(_In_ const WCHAR* wszMessage)
+{
+    HRESULT hr = S_OK;
+    WCHAR wszBuffer[LogEntryMaxSize] = _T("[IM:%s]");
```

Comment:
Can you initialize with wszLogFormattablePrefix? Or initially leave it empty and then append the const string?

Reply:
He can't directly initialize with wszLogFormattablePrefix because you can't initialize an array with a pointer. He could initialize with the const, if he wanted to use the tstring class, but that would incur allocation cost. The easiest thing to do to enable reuse is to use a macro.

---
In reply to: [333780025](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r333780025) [](ancestors = 333780025)

Reply:
I'll remove the wszLogFormattablePrefix and define a macro as Del suggested.

---
In reply to: [334116304](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r334116304) [](ancestors = 334116304,333780025)